### PR TITLE
mobile ucr caching

### PR DESCRIPTION
**Pull request by emord**
@snopoke you were right about oging this route. It ended up making the code easier to read I think. It also makes purging old reports that were deleted much easier

### CODE REVIEW

File: corehq/apps/app_manager/fixtures/mobile_ucr.py [hunk 1]
```diff
@@ -277,10 +278,67 @@ def __call__(self, restore_state):
...
+                fixtures.extend(self._empty_v2_fixtures(report_uuid))
 
         return fixtures
 
+    def _relevant_report_configs(self, restore_state, report_configs):
+        """
+        Filter out any UCRs that are already synced. This can't exist in V1,
+        because in V1 we send all reports as one fixture.
+
+        Returns a list of full ReportConfigs to sync and a set of report ids to purge
```
Comment (emord) [inline corehq/apps/app_manager/fixtures/mobile_ucr.py]: This feels a bit weird of a return value, but we need the full report config later and if an reports were deleted we only have the ID in the synclog

File: corehq/apps/app_manager/views/modules.py [hunk 2]
```diff
@@ -242,9 +243,10 @@ def _report_to_config(report):
             'moduleFilter': module.module_filter,
             'availableReports': [_report_to_config(r) for r in all_reports],  # structure for all reports
             'currentReports': [r.to_json() for r in module.report_configs],  # config data for app reports
-            'columnXpathTemplate': COLUMN_XPATH_CLIENT_TEMPLATE,
+            'columnXpathTemplate': get_column_xpath_client_template(app.mobile_ucr_restore_version),
```
Comment (emord) [inline corehq/apps/app_manager/views/modules.py]: I think this may have accidentally been left out of my earlier PR

**Review by snopoke — APPROVED** [review]
Nice

File: corehq/apps/app_manager/fixtures/mobile_ucr.py [hunk 3]
```diff
@@ -277,10 +280,67 @@ def __call__(self, restore_state):
...
+
+        config_uuids = {config.uuid for config in report_configs}
+        extra_configs_on_phone = set(last_ucr_syncs.keys()).difference(config_uuids)
+
+        return configs_to_sync, extra_configs_on_phone
+
+    def _empty_v2_fixtures(self, report_uuid):
+        return [
+            E.fixture(id=self._report_fixture_id(report_uuid)),
+            E.fixture(id=self._report_fixture_id(report_uuid))
```
[the remark this answers is not in the document]
Comment (emord) [reply corehq/apps/app_manager/fixtures/mobile_ucr.py]: Thanks I fixed it here https://github.com/dimagi/commcare-hq/pull/18370/commits/223e8c61c0604ba00d26aff407edb331bc8b7e7a started to write a test, but it looked like a refactor would be necessary before that unfortunately. (there are a lot of mocks and it doesn't use the db)

**Pull request merged**